java/semantickernel-api/src/main/java/com/microsoft/semantickernel/Kernel.java

```diff
@@ -59,25 +54,19 @@ public Kernel(
         this.globalKernelHooks = new KernelHooks(globalKernelHooks);
     }
 
-    public <T> Mono<FunctionResult<T>> invokeAsync(
-        KernelFunction<T> function,
-        @Nullable KernelFunctionArguments arguments) {
-        return invokeAsync(function, arguments, null);
-    }
-
-    public <T> Mono<FunctionResult<T>> invokeAsync(
-        KernelFunction<T> function,
-        @Nullable KernelFunctionArguments arguments,
-        @Nullable ContextVariableType<T> variableType) {
-        return invokeAsync(function, arguments, variableType, null);
+    public <T> FunctionInvocation<T> invokeAsync(
```

Comment:
Are you sure about not making KernelFunction fluent as well? It seems a little odd to have parallel API that isn't parallel.